### PR TITLE
Introduce PGP support with a passing test! ✅

### PR DESCRIPTION
**Pull request by mahmoudimus**
**Stack**:
- #287
- #286 ⮜
- #285
- #284
- #283
- #282
- #281
- #280
- #279
- #278
- #277
- #276
- #274
- #273
- #272
- #271
- #270
- #269
- #268
- #267
- #266
- #265
- #264
- #263
- #262
- #261
- #260
- #259

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

### CODE REVIEW

File: larky/src/main/resources/vendor/OpenPGP/__init__.star [hunk 1]
```diff
@@ -1697,468 +2030,288 @@ def _PublicKeyPacket():
...
+        if self.algorithm == 0:
+            self.data = Message.parse(self.data)
+        elif self.algorithm == 1:
+            self.data = Message.parse(zlib.decompress(self.data, -15))
+        elif self.algorithm == 2:
+            self.data = Message.parse(zlib.decompress(self.data))
+        elif self.algorithm == 3:
+            self.data = Message.parse(bz2.decompress(self.data))
+        else:
+            pass  # TODO: error?
```
Comment (jentfoo) [inline larky/src/main/resources/vendor/OpenPGP/__init__.star:2181]: Completing the TODO and providing errors would make debugging possible issues more obvious

**Review by jentfoo — COMMENTED** [review]
This looks pretty good, I do not see a lot to improve on, just a couple minor comments.  In general I think more tests would be a good idea, particularly tests focused around error and failure conditions.

File: larky/src/main/resources/vendor/OpenPGP/Crypto.star [hunk 2]
```diff
@@ -545,11 +391,157 @@ def _Wrapper():
...
+                    key_block_bytes, to_encrypt, lambda x: session_cipher.encrypt(x)
+                )
+            )
+        ]
+
+        if not types.is_iterable(passphrases_and_keys) or types.is_string(passphrases_and_keys):
+            passphrases_and_keys = [passphrases_and_keys]
+
+        for psswd in passphrases_and_keys:
+            print("password: ", psswd)
```
[the remark this answers is not in the document]
Comment (mahmoudimus) [reply larky/src/main/resources/vendor/OpenPGP/Crypto.star:427]: Addressed here: https://github.com/verygoodsecurity/starlarky/pull/287/files

File: larky/src/main/resources/vendor/OpenPGP/__init__.star (hunk 1 above)
Comment (mahmoudimus) [reply larky/src/main/resources/vendor/OpenPGP/__init__.star:2181]: Copied from the original stuff: https://github.com/mahmoudimus/OpenPGP-Python/blob/fork/src/OpenPGP/__init__.py#L1608-L1638

**Review by aslepakurov — APPROVED** [review]

**Pull request merged**